compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaServiceDeclarationSymbol.java

```diff
@@ -59,6 +62,7 @@
     private final TypeSymbol typeDescriptor;
     private final ServiceAttachPoint attachPoint;
 
+    private Set<TypeSymbol> listenerTypes;
```

Comment:
Shall we change this to List? if have multiple expressions with the same type reference we will get only one record. Which is not correct IMO.

Reply:
Wouldn't that be redundant? For example,
```ballerina
listener http:Listener foo  = new(9090);
listener http:Listener bar  = new(8080);

service / on foo, bar {
}
```
In the above, if we return a list, it'll be `[http:Listener, http:Listener]`. Isn't this repetitive info that doesn't really add any additional insight?

Reply:
This can mislead those who use the API. Sometimes we need to related expression with the type. this will break this. Example. 

```
listener http:Listener foo  = new(9090);
listener http:Listener2 bar  = new(8080);
listener http:Listener baz  = new(8088);

service / on foo, bar, baz{
}
```

Above giving [http:Listener, http:Listener2] is wrong.

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/29614/commits/a5708ea50a63172b7420097e93abcdea7e289aa0